test/resource_test.rb

```diff
@@ -249,6 +249,34 @@ def test_quota_minimum_one_ticket
     assert_equal 1, resource.tickets
   end
 
+  def test_switch_static_tickets_to_quota
+
+    workers = 50
+    quota = 0.5
+
+    # Fork a large number of workers using static ticket strategy
+    fork_workers(count: workers - 1, tickets: 5, timeout: 0.1, wait_for_timeout: true) do
+      sleep 1
+    end
+
+    # Signal static workers to shut down
+    signal_workers('TERM')
+
+    # Create a quota based worker, and ensure it accounts for the static
+    # workers that haven't shut down yet
+    resource = create_resource :testing, quota: quota, timeout: 0.1
+    assert_equal((quota * workers).ceil, resource.tickets)
```

Comment:
Note that this points out an edge case that already exists - when the number of workers decreases, we will always have more tickets than the quota would specify *until* new workers boot.

This should be acceptable in practice however, as the number of tickets will be corrected later during the same deploy as more quota workers boot, or at the very latest at the start of the next deploy.